db/version_set.h

```diff
@@ -776,6 +777,13 @@ class Version {
     return storage_info_.user_comparator_;
   }
 
+  // Interprets *value as a blob reference, and (assuming the corresponding
+  // blob file is part of this Version) retrieves the blob and saves it in
+  // *value, replacing the blob reference.
+  // REQUIRES: *value stores an encoded blob reference
+  Status GetBlob(const ReadOptions& read_options, const Slice& user_key,
```

Comment:
Any reason why use `value` as both input and output argument?

Reply:
It minimizes data movement. `value` here is the actual `PinnableSlice` passed in by the user. It makes sense for `Version::Get` to pass it (in `GetContext`) to `TableCache::Get` to eliminate additional copies/moves in the common case when the value is not a `BlobIndex`. On the other hand, when it *is* a `BlobIndex`, the value read from the table (which is in `value` at that point) becomes the input for `GetBlob`, and I feel it again makes sense to use it as output as well to eliminate the need for any further copying/moving. (In other words, if we were to use separate input and output arguments for `GetBlob`, we would eventually need to move `output_value` back to `value` at the call site.)